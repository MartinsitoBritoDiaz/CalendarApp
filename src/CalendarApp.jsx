import { BrowserRouter } from "react-router-dom";
import { AppRouter } from "./router";

export const CalendarApp = () => {
  return (
    <div className="main-container">
      <BrowserRouter>
        <AppRouter />
      </BrowserRouter>
    </div>
  );
};
