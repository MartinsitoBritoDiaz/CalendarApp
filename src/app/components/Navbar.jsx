export const Navbar = () => {
  return (
    <nav className="navbar custom--navbar" >
      <div className="container-fluid d-flex justify-content-between">
        <div>
          <a className="navbar-brand mt-2 mt-lg-0" href="#">
            <span className="fw-bold text-light">CalendarApp</span>
          </a>
        </div>

        <div className="d-flex align-items-center">
          <a className="link-secondary me-3 text-light" href="#" role="button">
            <i className="fa-solid fa-right-from-bracket"></i>
          </a>
        </div>
      </div>
    </nav>
  );
};
