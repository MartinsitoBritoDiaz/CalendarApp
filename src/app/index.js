export * from './pages/CalendarPage'
export * from './routes/CalendarRoutes'