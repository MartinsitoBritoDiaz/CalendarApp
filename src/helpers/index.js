export * from './calendarLocalizer';
export * from './customStyles';
export * from './getEnvVariables';
export * from './getMessages';
export * from './convertEventsToDateEvents';