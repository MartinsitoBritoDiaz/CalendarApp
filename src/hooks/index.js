export * from './useAuthStore';
export * from './useAuthVerification';
export * from './useCalendarStore';
export * from './useCustomForm';
export * from './useForm';
export * from './useUIStore';