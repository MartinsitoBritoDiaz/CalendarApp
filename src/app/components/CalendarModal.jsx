import "sweetalert2/dist/sweetalert2.min.css";

import Modal from "react-modal";

import "react-datepicker/dist/react-datepicker.css";
import DatePicker from "react-datepicker";
import { useUIStore } from "../../hooks";
import { customStyles } from "../../helper";
import { useCustomForm } from "../../hooks";


Modal.setAppElement("#root");

export function CalendarModal() {

  const {titleClass, formValues, onSubmit, updateFormValues, isDateModalOpen, closeDateModal } = useCustomForm();

  const onInputChange = ({ target }) => {
    updateFormValues({
      ...formValues,
      [target.name]: target.value,
    });
  };

  const onDateChange = (event, changing) => {
    updateFormValues({
      ...formValues,
      [changing]: event,
    });
  };

  const onCloseModal = () => {
    closeDateModal();
  };

  return (
    <Modal
      isOpen={isDateModalOpen}
      onRequestClose={onCloseModal}
      style={customStyles}
      className="modal"
      overlayClassName="modal-fondo"
      closeTimeoutMS={200}
    >
      <div className="d-flex justify-content-between">
        <h1 className="fs-3 my-3"> New Event </h1>
        <button className="btn btn-lg btn--closeModal" onClick={onCloseModal}>
          <i className="fas fa-xmark"></i>
        </button>
      </div>

      <form className="container " onSubmit={onSubmit}>
        <div className="form-group mb-2">
          <label>Start date</label>
          <DatePicker
            selected={formValues.start}
            className="form-control"
            dateFormat="Pp"
            showTimeSelect
            onChange={(event) => onDateChange(event, "start")}
          />
        </div>

        <div className="form-group mb-2">
          <label>End date</label>
          <DatePicker
            minDate={formValues.start}
            selected={formValues.end}
            className="form-control"
            dateFormat="Pp"
            showTimeSelect
            onChange={(event) => onDateChange(event, "end")}
          />
        </div>

        <hr />
        <div className="form-group mb-2">
          <label>Title</label>
          <input
            type="text"
            className={`form-control ${titleClass}`}
            placeholder="Birthday"
            name="title"
            autoComplete="off"
            value={formValues.title}
            onChange={onInputChange}
          />
          <small id="emailHelp" className="form-text text-light">
            Short description
          </small>
        </div>

        <div className="form-group mb-2">
          <textarea
            type="text"
            className="form-control"
            placeholder="Notes"
            rows="5"
            name="notes"
            value={formValues.notes}
            onChange={onInputChange}
          ></textarea>
          <small id="emailHelp" className="form-text text-muted">
            Aditional information
          </small>
        </div>

        <div className="d-flex justify-content-end">
          <button type="submit" className="btn btn--custom btn-block align-left">
            <i className="far fa-save"></i>
            <span> Save</span>
          </button>
        </div>
      </form>
    </Modal>
  );
};
