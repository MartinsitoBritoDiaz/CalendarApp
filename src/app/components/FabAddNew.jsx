import React from "react";
import { useCalendarStore, useUIStore } from "../../hooks";
import { addHours } from "date-fns";

export function FabAddNew() {
  const { openDateModal } = useUIStore();
  const { setActiveEvent } = useCalendarStore();

  const handleCLickNew = () => {
    setActiveEvent({
      title: "",
      start: new Date(),
      notes: "",
      end: addHours(new Date(), 2),
      bgColor: "#fafafa",
      user: {
        _id: "123",
        name: "Luisin",
      },
    });
    openDateModal();
  };
  return (
    <button className="btn btn-secondary fab btn--custom" onClick={handleCLickNew}>
      <i className="fas fa-duotone fa-plus"></i>
    </button>
  );
}
