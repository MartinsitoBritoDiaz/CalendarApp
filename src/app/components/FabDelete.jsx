import { useCalendarStore, useUIStore } from "../../hooks";

export function FabDelete() {
  const { startDeletingEvent, hasEventSelected } = useCalendarStore();

  const handleDelete = () => {
    startDeletingEvent();
  };
  return (
    <button
      aria-label="btn-delete"
      className="btn btn-danger fab-delete btn--custom__red"
      onClick={handleDelete}
      style={{
        display: hasEventSelected ? "" : "none",
      }}
    >
      <i className=" fas fa-trash-can"></i>
    </button>
  );
}
