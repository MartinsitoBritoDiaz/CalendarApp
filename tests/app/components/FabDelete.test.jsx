/**
 * @jest-environment jsdom
 */
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { FabDelete } from '../../../src/app/components/FabDelete';
import { store } from '../../../src/store';
import { useCalendarStore } from '../../../src/hooks';

jest.mock('../../../src/hooks/useCalendarStore');

describe('Testing inside <FabDelete />', () => {
  test('should render the button', () => {
    useCalendarStore.mockReturnValue({
      hasEventSelected: false
    })

    render(
        <FabDelete />
    );

    const btn = screen.getByLabelText('btn-delete');

    expect( btn.style.display ).toBe('none');

    screen.debug();
  });
});
