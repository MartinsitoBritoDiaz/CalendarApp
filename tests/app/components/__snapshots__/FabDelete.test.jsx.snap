// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing inside <FabDelete /> should render the button 1`] = `
<div>
  <button
    aria-label="btn-delete"
    class="btn btn-danger fab-delete btn--custom__red"
    style="display: none;"
  >
    <i
      class=" fas fa-trash-can"
    />
  </button>
</div>
`;
